pymongo/server.py

```diff
@@ -132,6 +132,8 @@ def run_operation(
 
         if publish:
             cmd, dbn = operation.as_command(conn)
+            if "$db" not in cmd and dbn is not None:
```

Comment:
2 questions: Is dbn ever None here? Could we move this logic to as_command?

Reply:
I still have this question.

Reply:
`dbn` does not appear to ever be `None`, but moving this logic to `as_command` does not work, the test fails with the same "missing $db" error message.